src/components/MenuSection/MenuSection.jsx

```diff
@@ -0,0 +1,89 @@
+/* eslint-disable react-hooks/exhaustive-deps */
+/* eslint-disable no-unused-vars */
+/* eslint-disable react/prop-types */
+import { useEffect, useState } from 'react';
+import { FlexContainer } from '../FlexContainer/FlexContainer';
+import ProductCard from '../Card/Card';
+import Button from '../Button/Button';
+import BrowseMenuText from '../BrowseMenuText/BrowseMenuText';
+import ToggleButtons from '../ToggleButtons/ToggleButtons';
+import './MenuSection.css';
+
+const MenuSection = ({ addToCart }) => {
+    const [products, setProducts] = useState([]);
```

Comment:
I would suggest to think about so many states here. It's better to define state value only if it's really needed as evere state change lead as to rerender component